### ISSUE TITLE
会重复收到消息

### ISSUE THREAD

**Issue by hgncxzy** [JeremyLiao/LiveEventBus#122]
我发现会重复收到消息，版本是v1.7.2,传的是 map，语言时 kotlin

**Comment by JeremyLiao** [comment]
可以先用console功能debug一下

**Comment by lzy-h2o2** [comment]
我也遇到类似的问题  点击一次按钮  触发多次，发现version在自增：
2020-12-17 16:28:32.021 26103-26103/com.app.base I/xxx: ==> *********Base info*********
    lifecycleObserverAlwaysActive: false
    autoClear: true
    logger enable: true
    logger: com.zndroid.bus.logger.DefaultLogger@6b10207
    Receiver register: true
    Application: android.app.Application@c12775e
    *********Event info*********
    Event name: com.zndroid.base.event.BaseNetEvent
    	version: 3 ========>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>这里
    	hasActiveObservers: true
    	hasObservers: true
    	ActiveCount: 1
    	ObserverCount: 1
    	Observers: 
    		[com.zndroid.bus.core.LiveEventBusCore$ObserverWrapper@6983ac3=androidx.lifecycle.ExternalLiveData$ExternalLifecycleBoundObserver@562d672]


我的触发时机是：日/夜间模式来回切换
调用了 recreate();方法

**Comment by hgncxzy** [comment]
我这个问题，最后定位不是框架问题，是两个app之间的全局广播导致的。












许助云
hgncxzy@qq.com


---原始邮件---
发件人: "lazy"<notifications@github.com&gt;
发送时间: 2020年12月17日(周四) 下午4:32
收件人: "JeremyLiao/LiveEventBus"<LiveEventBus@noreply.github.com&gt;;
抄送: "Author"<author@noreply.github.com&gt;;"hgncxzy"<hgncxzy@qq.com&gt;;
主题: Re: [JeremyLiao/LiveEventBus] 会重复收到消息 (#122)




 
我也遇到类似的问题  点击一次按钮  触发多次，发现version在自增：
 2020-12-17 16:28:32.021 26103-26103/com.app.base I/xxx: ==&gt; Base info
 lifecycleObserverAlwaysActive: false
 autoClear: true
 logger enable: true
 logger: com.zndroid.bus.logger.DefaultLogger@6b10207
 Receiver register: true
 Application: android.app.Application@c12775e
 Event info
 Event name: com.zndroid.base.event.BaseNetEvent
 version: 3 ========&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;这里
 hasActiveObservers: true
 hasObservers: true
 ActiveCount: 1
 ObserverCount: 1
 Observers:
 [com.zndroid.bus.core.LiveEventBusCore$ObserverWrapper@6983ac3=androidx.lifecycle.ExternalLiveData$ExternalLifecycleBoundObserver@562d672]
 
我的触发时机是：日/夜间模式来回切换
 调用了 recreate();方法

**Comment by lzy-h2o2** [comment]
仔细观察了一下  是recreate()导致的 fragment叠加导致的

**Comment by JeremyLiao** [comment]
> version

version自增是正常的，ObserverCount=1感觉也没有问题，最后是fragment叠加导致的是吧？

**Comment by lzy-h2o2** [comment]
是的 luzhenyuxfcy 邮箱：luzhenyuxfcy@sina.com 签名由 网易邮箱大师 定制 在2020年12月17日 18:39，Jeremy Liao 写道： version version自增是正常的，ObserverCount=1感觉也没有问题，最后是fragment叠加导致的是吧？ — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.
